src/Symfony/Component/Validator/Violation/ConstraintViolationBuilder.php

```diff
@@ -73,7 +73,7 @@ public function setParameters(array $parameters): static
         return $this;
     }
 
-    public function setTranslationDomain(string $translationDomain): static
+    public function setTranslationDomain(string|false|null $translationDomain): static
```

Comment:
BC break. I'd suggest reverting and allow false to be passed only via constructor.